pandas/core/algorithms.py

```diff
@@ -467,6 +467,8 @@ def isin(comps: AnyArrayLike, values: AnyArrayLike) -> np.ndarray:
     elif is_extension_array_dtype(comps.dtype) or is_extension_array_dtype(
         values.dtype
     ):
+        if type(comps).__name__ == "IntegerArray":
+            comps = comps._data  # type: ignore[attr-defined, assignment]
```

Comment:
This is going to lead to incorrect answers, e.g.

```
import pandas as pd
from pandas.core.algorithms import isin

arr = pd.array([2, 3, pd.NA, 5])

>>> isin(arr, [1])
array([False, False, False, False])
>>> isin(arr._data, [1])
array([False, False,  True, False])
```

Reply:
Hi @jbrockmendel, thanks for pointing out this. If there is pd.NA in the array, the result could be incorrect, but we can multiply the result with ~comps._mask, which is a boolean numpy array:
```Python
def isin_for_masked_array(comps, values):
    if isinstance(comps, BaseMaskedArray):
        _comps = comps._data
        result = isin(_comps, values) * np.invert(comps._mask)
        return result
    return isin(comps, values)
```
I've tested the result is correct and the runtime is not bad.

Reply:
And if the case is isin([pd.NA], [pd.NA]), then above solution is not correct. We have to check if there is null value in the second array.

Reply:
> We have to check if there is null value in the second array

Have to be a little bit careful about which null values you allow, e.g. if values contains `pd.NaT` and comps is an IntegerArray, we probably dont want to count it.

As for whether to set NA locations to False or NA, best to check with @jorisvandenbossche for his thoughts on the desired behavior.

On the code organization, #38422 is a sketch of where I suggest you put the implementation.

Reply:
@jbrockmendel @jorisvandenbossche Hi, I guess `isin(1, 2, pd.NA], [1, pd.NA])` could return `[True, False, True]` but not `[True, False, pd.NA]`. 
Below are something might be useful, and I am also curious why `pd.NA in pd.Series([1, 2, pd.NA]` should return False?
```Python
None in [1, 2, 3, None]  # Will return True but not None
np.nan in np.array([1, 2, np.nan])  # Will return False but not np.nan
pd.NA in pd.Series([1, 2, pd.NA])  # Will return False but not pd.NA
```
Would you like to put some thoughts here?